src/DynamoCore/Library/LibraryServices.cs

```diff
@@ -95,6 +114,45 @@ private void PreloadLibraries()
                 CompilerUtils.TryLoadAssemblyIntoCore(LibraryManagementCore, library);
         }
 
+        public bool FunctionSignatureNeedsAdditionalAttributes(string functionSignature)
+        {
+            return priorNameHints[functionSignature].AdditionalAttributes.Count > 0;
```

Comment:
Do a `KeyExists` check on `priorNameHints` before trying to get the `AdditionalAttributes`.